### PR TITLE
Add new endpoint and deserializer

### PR DESCRIPTION
**Pull request by patri-create**
### :pushpin: References
* **Issue:** [Improve networking performance](https://github.com/Sefford/Artdrian/issues/27)

### :tophat: What is the goal?

Add the new endpoint to make the API works again.

### How is it being implemented?

The new endpoint has a different structure, to accommodate that, I have used a JSON deserializer. I have wrapped the new API response into a WallpaperResponse object. I have fixed temporarily some parts of the code base.

In the upcoming PR, I'll add DTOs and their mapping, as well as fix tests.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -78,6 +78,7 @@ dependencies {
 
     implementation 'com.squareup.retrofit2:retrofit:2.9.0'
     implementation 'com.squareup.retrofit2:converter-gson:2.9.0'
+    implementation "com.squareup.okhttp3:logging-interceptor:4.9.3"
```
Comment (patri-create) [inline app/build.gradle:81]: I added a logger in the network layer. I used this to debug the calls.

File: app/src/androidTest/java/com/sefford/artdrian/di/FakeApiModule.kt [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
 class FakeApiModule {
 
     @Provides
     @Singleton
-    fun providesWallpaperApi(): WallpaperApi = FakeWallpaperApi{ WALLPAPER_LIST.map { it.metadata } }
+    fun providesWallpaperApi(): WallpaperApi = FakeWallpaperApi{ WallpaperResponse(WALLPAPER_LIST.map { it.metadata }) }
+
+    @Provides
+    @Singleton
+    fun providesCoroutineDispatcher() = Dispatchers.Unconfined
```
Comment (patri-create) [inline app/src/androidTest/java/com/sefford/artdrian/di/FakeApiModule.kt:21]: This fixes some tests.

File: app/src/main/java/com/sefford/artdrian/model/Metadata.kt [hunk 3]
```diff
@@ -4,13 +4,13 @@ import com.google.gson.annotations.SerializedName
 import java.util.Date
 
 data class Metadata(
-  @SerializedName("ID") val id: String,
-  @SerializedName("Slug") val slug: String,
+  @SerializedName("_id") val id: String,
```
Comment (patri-create) [inline app/src/main/java/com/sefford/artdrian/model/Metadata.kt:7]: I fix the rest of the params in the next PR.

File: app/src/test/java/com/sefford/artdrian/data/dto/deserializers/WallpaperResponseDeserializerTest.kt [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import org.junit.jupiter.api.Test
+
+class WallpaperResponseDeserializerTest : Files {
+
+    private val gson : Gson = GsonBuilder()
+        .registerTypeAdapter(WallpaperResponse::class.java, WallpaperResponseDeserializer())
+        .create()
+
+    @Test
+    fun deserialize() {
```
Comment (patri-create) [inline app/src/test/java/com/sefford/artdrian/data/dto/deserializers/WallpaperResponseDeserializerTest.kt:26]: Is it ok this test for the deserializer? 🤔

File: app/src/test/resources/com/sefford/artdrian/data/dto/deserializers/index.json [hunk 5]
```diff
@@ -0,0 +1 @@
...i of l(t))!w.call(e,i)&&i!==a&&n(e,i,{get:()=>t[i],enumerable:!(r=c(t,i))||r.enumerable});return e};var x=(e,t,a)=>(a=e!=null?_(u(e)):{},o(t||!e||!e.__esModule?n(a,\"default\",{value:e,enumerable:!0}):a,e)),b=e=>o(n({},\"__esModule\",{value:!0}),e);var m=d((C,g)=>{g.exports=_jsx_runtime});var y={};h(y,{default:()=>f,frontmatter:()=>j});var s=x(m()),j={title:\"Ghost waves\",edition:\"001\",publishedAt:\"2022-08-01\",description:\"\",className:\"bg-wallpaper-ghost_waves_001 hover:bg-wallpaper-ghost_waves_001\",image:\"/static/images/wallpaper/ghost_waves_001_web_preview.jpg\",imageObjectPosition:\"70% 50%\",imageUnoptimized:!0,imageQuality:100,tags:[\"4k-ready\"]};function p(e){return(0,s.jsx)(s.Fragment,{})}function v(e={}){let{wrapper:t}=e.components||{};return t?(0,s.jsx)(t,Object.assign({},e,{children:(0,s.jsx)(p,e)})):p(e)}var f=v;return b(y);})();\n;return Component;"},"_id":"wallpapers/ghost_waves_001.mdx","_raw":{"sourceFilePath":"wallpapers/ghost_waves_001.mdx","sourceFileName":"ghost_waves_001.mdx","sourceFileDir":"wallpapers","contentType":"mdx","flattenedPath":"wallpapers/ghost_waves_001"},"type":"Wallpaper","slug":"ghost_waves_001","totalDownloads":992}]},"__N_SSG":true}
```
Comment (patri-create) [inline app/src/test/resources/com/sefford/artdrian/data/dto/deserializers/index.json:1]: I added the response json into a file.

File: app/build.gradle (hunk 1 above)
Comment (Sefford) [reply app/build.gradle:81]: Classic!

File: app/src/test/java/com/sefford/artdrian/data/dto/deserializers/WallpaperResponseDeserializerTest.kt (hunk 4 above)
Comment (Sefford) [reply app/src/test/java/com/sefford/artdrian/data/dto/deserializers/WallpaperResponseDeserializerTest.kt:26]: You know I am very succint for testing names. It's just a name, but can be improved. Maybe we can talk about this on our next 1:1.

**Review by Sefford — APPROVED** [review]
Great job overall!

**Pull request merged**